includes/Core/Authentication/Google_Proxy.php

```diff
@@ -85,15 +85,16 @@ public function get_site_fields() {
 		$home_url_no_scheme = str_replace( array( 'http://', 'https://' ), '', home_url() );
 
 		return array(
-			'name'       => wp_specialchars_decode( get_bloginfo( 'name' ) ),
-			'url'        => home_url(),
-			'action_uri' => admin_url( 'index.php' ),
-			'return_uri' => $this->context->admin_url( 'splash' ),
+			'name'         => wp_specialchars_decode( get_bloginfo( 'name' ) ),
+			'url'          => home_url(),
+			'action_uri'   => admin_url( 'index.php' ),
+			'return_uri'   => $this->context->admin_url( 'splash' ),
 			// TODO: Remove admin_root once proxy is updated.
-			'admin_root' => str_replace( array( 'http://', 'https://', $home_url_no_scheme ), '', admin_url() ),
+			'admin_root'   => str_replace( array( 'http://', 'https://', $home_url_no_scheme ), '', admin_url() ),
+			'redirect_uri' => add_query_arg( 'oauth2callback', 1, admin_url( 'index.php' ) ),
```

Comment:
This is now duplicate logic, since it doesn't use the `OAuth_Client::get_redirect_uri()`. It doesn't have access to this, so we'll need to restructure a bit so that all relevant areas can get that redirect URI. That shouldn't be a blocker for now though since this is time-sensitive.

cc @ryanwelcher @aaemnnosttv If you have suggestions for how to restructure the code to have a single place for the redirect URI, let's open a follow-up issue.